server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -150,4 +188,80 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         });
 
     }
+
+    private void validate(
+        ClusterState currentState,
+        Map<String, Tuple<Boolean, IndexMetadata>> indexMetadataMap,
+        boolean restoreAllShards,
+        String restoreClusterUUID
+    ) {
+        String errorMsg = "cannot restore index [%s] because an open index with same name/uuid already exists in the cluster.";
+
+        List<String> graveyardIndexNames = new ArrayList<>();
+        List<String> graveyardIndexUUID = new ArrayList<>();
```

Comment:
use a HashMap with IndexName as key and UUID as value.  Name & UUID together should be unique. Also hashmap look ups will be faster than List.contains

Reply:
if we do name & uuid together unique instead of individually then it can lead to problem as same name index creation is not allowed and same uuid index wont be allowed as well.